src/prefect/client/client.py

```diff
@@ -139,60 +128,20 @@ def __init__(
             or cached_auth.get("api_key")
         )
 
+        # mypy struggles with this attribute type if not created here
+        self._tenant_id: Optional[str] = None
+
         # Load the tenant id
-        self._tenant_id: Optional[str] = (
-            tenant_id
+        # This assignment is validated by the `Client.tenant_id` setter
+        self.tenant_id: Optional[str] = (
```

Comment:
how do we infer a tenant_id from an api key again?
does the user have to specify a tenant?

Reply:
answering my own question again nvm I see the logic in `_get_auth_tenant` now